### PR TITLE
feat(ui): UI/UX overhaul phases 1-2 — default polish and reasoning visibility

### PR DESCRIPTION
**Pull request by lvndry**
## What this PR delivers

The first two phases of the UI/UX overhaul that came out of a five-subagent flaw hunt across the Jazz terminal UI (55 flaws catalogued). Phase 1 makes the interface feel alive and honest — the polished unicode chrome ships by default, every waiting state shows real elapsed time, and the input line stops fighting the user. Phase 2 makes reasoning a first-class, recoverable artifact instead of ephemeral text that vanishes on collapse or interrupt.

It also fixes the reported bug where the slash-command suggestion dropdown stayed painted on screen for the whole duration of the agent's response.

## Why this matters

For users: the CLI previously shipped ASCII `+ | - o ?` chrome even in terminals that render `✓ ✗ ❯ ⠋` perfectly; long tool runs and thinking phases were indistinguishable from hangs (no elapsed time anywhere); typing `/clear` + Enter did nothing until a second Enter; yolo mode became invisible 2 seconds after toggling it — a genuine safety hazard; and reasoning older than the last block was permanently unrecoverable.

For maintainers: the worst raw-glyph literals (`✔ ◆ ⛁⛶⛝ ● •`) now route through the `glyphs.ts` module that existed precisely to gate them, so ASCII terminals stop getting tofu and future glyph changes have one home.

## How UI behaviour changes

| Surface | Change |
|---|---|
| Glyph mode | Auto-detects unicode (UTF-8 locale, known terminal, or `WT_SESSION`); falls back to ASCII on `TERM=dumb`/`linux` or no signal. `JAZZ_UI_GLYPHS=ascii\|unicode` still overrides both ways. Pinned by `glyphs.test.ts`. |
| Slash suggestions | Dropdown frame is erased on prompt dismissal (Ink instance `clear()` via new `registerFrameClearHandler`); dropdown capped at 8 rows with an `…and N more` overflow (tall frames were what broke Ink's erase). |
| Enter on a command | `/clear` + Enter runs on the first Enter when the typed value exactly matches a command; incomplete prefixes still complete in place. Tab now accepts the highlighted suggestion (was a dead key). |
| Waiting states | Awaiting/thinking/streaming/tool-execution all show a live `· 12s` elapsed counter (appears after 2s); ephemeral panel headers self-tick every second instead of freezing when content stalls; the 30s slow-tool warning re-arms periodically and names the interrupt shortcut. |
| Streamed turns | Each turn starts with a `◆ AgentName (provider/model)` agent-colored header instead of a generic info line. Settled stream slices render pre-wrapped (`PreWrappedText`), closing the remaining Yoga char-by-char wrap path. |
| Footer | Persistent `yolo` indicator (warning-colored, bold) whenever auto-approve is on; context shows `12.3k/200k` — the denominator resolves from the models.dev cache at stream start (was dead code, never populated); stats line uses readable secondary color instead of dim. |
| Reasoning | Live panel renders dim/italic (matches settled styling — no more bright-then-dim jump); panel height adapts to terminal rows (¼ of rows, clamped 8–24, was fixed 8); collapsed blocks accumulate in a 20-deep stack so repeated Ctrl+R walks back through a whole multi-step turn (was single slot, earlier blocks lost); collapse line now says `· ctrl+r to expand`. |
| Interrupt (double-Esc) | Prints `Interrupting…` immediately; open reasoning is preserved into the Ctrl+R stack instead of destroyed; subagent panels collapse on interruption via `Effect.onInterrupt` (previously orphaned on non-Esc aborts). |
| Tool results | Lines carrying their own ANSI styling (diff +/- colors, syntax highlighting) are no longer wrapped in `dimColor`; truncated results of any tool are expandable with Ctrl+O (was diff-only) and the truncation notice says so. |
| `/context` grid | Uses width-safe `█ ░ ▒` (unicode) / `# . ~` (ascii) instead of ambiguous-width `⛁ ⛶ ⛝`. |

## UX changes operators will notice

- Out of the box on a modern terminal: braille spinner, `✓ ✗ ⚠ •` status marks, `❯` prompt cursor.
- Your own messages read `❯ You:` in bold primary color — scannable when scrolling back.
- Every "the agent is doing something" state has a visible, advancing timer.
- The footer always tells you when you're in yolo mode and how full the context window is.
- After a turn with several reasoning phases, pressing Ctrl+R repeatedly re-emits each block, newest first, labeled with its duration.

## Developer-visible API changes

| Surface | Old | New |
|---|---|---|
| `GlyphSet` | — | new fields `active`, `diamond`, `gridFilled`, `gridEmpty`, `gridReserved` |
| `store` | `setExpandableReasoning` single slot | bounded stack (`MAX_EXPANDABLE_REASONING = 20`); `expandLastReasoning()` pops |
| `store` | — | `registerFrameClearHandler(handler)` + `registerModeSetter(setter)` |
| `tool-formatter` | — | exported `expandableToolResultPayload(result)` |
| `stream_start` output | `info` entry (plain string) | `log` entry (ink node header) |

## Tests

- 1414 passing across 83 files (5 new tests; 4 existing tests updated to match the new header/glyph/expansion contracts).
- `src/cli/ui/glyphs.test.ts` — canonical pin for the detection matrix (override, UTF-8 locale, TERM_PROGRAM allowlist, WT_SESSION, dumb/linux fallback).
- `src/cli/ui/store.test.ts` — pins the reasoning stack walk-back (`expandLastReasoning walks back through multiple collapsed blocks`).
- `src/cli/presentation/ink-presentation-service.test.ts` — pins that streaming never renders the response card (now content-based, tolerant of the turn header).

### Edge cases to verify in a staging session

1. Type `/` while an agent is responding, then send — expected: no suggestion rows linger above the streaming output.
2. Type `/clear` and press Enter once — expected: screen clears immediately (no second Enter needed).
3. `JAZZ_UI_GLYPHS=ascii jazz …` in a UTF-8 terminal — expected: pure ASCII chrome everywhere, including todo checklists and the `/context` grid.
4. Run a tool that takes >60s — expected: warning lines at ~30s and ~60s naming the Esc-Esc interrupt, elapsed counter advancing next to the tool label.
5. Interrupt mid-reasoning with double-Esc — expected: `Interrupting…` prints at once and Ctrl+R afterwards recovers the partial reasoning tail.
6. Toggle yolo via Shift+Tab and wait out the toast — expected: `yolo` stays visible in the footer until toggled back.

## Notes for reviewers

- The suggestion-dropdown fix is terminal-plumbing, not React state: `setPrompt(null)` already unmounted the dropdown synchronously, but Ink doesn't reliably erase lines freed by a shrinking live region under heavy `<Static>` churn. The fix erases the painted frame (`inkInstance.clear()`) at dismissal time; the 8-row dropdown cap reduces the trigger surface.
- `collapseAllEphemeral()` preserves only the visible tail of an interrupted reasoning region (the full text lives in the renderer, which is being torn down at that moment) — partial recovery is intentional, not an oversight.
- The un-dim heuristic in the tool-result card checks each line for the ANSI escape prefix (`ESC [`) — any line already styled renders as-is; unstyled lines keep the dim treatment.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/cli/ui/Prompt.tsx [hunk 1]
```diff
@@ -257,21 +278,38 @@ function PromptComponent({
...
+                      />
+                    ))}
+                    {hiddenBelow > 0 && (
+                      <Box marginLeft={1}>
+                        <Text dimColor> …and {hiddenBelow} more</Text>
+                      </Box>
+                    )}
+                  </Box>
+                );
+              })()}
```
[the remark this answers is not in the document]
Comment (lvndry) [reply src/cli/ui/Prompt.tsx]: Done in 89997ba — the window math (`suggestionWindowStart`, `visibleSuggestions`, `hiddenSuggestionsBelow`) is now computed at the top of the component and the IIFE is gone.

File: src/cli/ui/store.ts [hunk 2]
```diff
@@ -435,30 +463,51 @@ export class UIStore {
...
     this.printOutput({
       type: "streamContent",
-      message: value.fullText,
+      message: `*${value.label} · ${seconds}s*\n\n${value.fullText}`,
       meta: { kind: "reasoning" },
       timestamp: new Date(),
     });
-    this.setExpandableReasoning(null);
+    this.setExpandableReasoning(this.expandableReasoningStack.at(-1) ?? null);
   };
```
[the remark this answers is not in the document]
Comment (lvndry) [reply src/cli/ui/store.ts:511]: Fixed in 89997ba — `clearOutputs()` now empties `expandableReasoningStack` and resets the footer indicator, so Ctrl+R can't resurrect reasoning from before a `/clear`. Cross-turn recall within an uncleared session is intentional (the stack is the recovery mechanism for interrupted turns).

File: src/core/utils/tool-formatter.ts [hunk 3]
```diff
@@ -362,6 +362,26 @@ function formatLoadSkillSectionStringBody(body: string): string {
...
+  try {
+    pretty = JSON.stringify(JSON.parse(normalized), null, 2);
+  } catch {
+    // Not JSON — expand the raw text as-is.
+  }
+  const withinCaps =
+    pretty.split("\n").length <= MAX_RESULT_DISPLAY_LINES &&
+    pretty.length <= MAX_RESULT_DISPLAY_CHARS;
+  return withinCaps ? null : pretty;
+}
```
[the remark this answers is not in the document]
Comment (lvndry) [reply src/core/utils/tool-formatter.ts:390]: Applied in 89997ba with one change from the suggestion: the 100k cap is applied *after* JSON pretty-printing rather than before, since truncating first would corrupt valid JSON and disable the pretty-print path entirely.

File: src/cli/presentation/ink-presentation-service.ts [hunk 4]
```diff
@@ -704,16 +725,43 @@ export class InkStreamingRenderer implements StreamingRenderer {
     }
```
[the remark this answers is not in the document]
Comment (lvndry) [reply src/cli/presentation/ink-presentation-service.ts:725]: This is already implemented exactly as described: `toolTimeouts` is a `Map<string, timeout>` (line 113), `clearToolTimeout(toolCallId)` cancels the specific timer on `tool_execution_complete` (line 490), and `clearAllToolTimeouts()` runs on both teardown paths — `handleComplete` and `flush` (lines 311/329). The re-armed timer at line 761 re-registers into the same map, so it's covered by the same cleanup. No change needed.
